dvc/command/pipeline.py

```diff
@@ -71,23 +71,46 @@ def __build_graph(self, target, commands, outs):
             else:
                 edges.append((from_stage.relpath, to_stage.relpath))
 
-        return nodes, edges
+        return nodes, edges, networkx.is_tree(G)
 
     def _show_ascii(self, target, commands, outs):
         from dvc.dagascii import draw
 
-        nodes, edges = self.__build_graph(target, commands, outs)
+        nodes, edges, _ = self.__build_graph(target, commands, outs)
 
         if not nodes:
             return
 
         draw(nodes, edges)
 
+    def _show_dependencies_tree(self, target, commands, outs):
+        from treelib import Tree
+
+        nodes, edges, is_tree = self.__build_graph(target, commands, outs)
+        if not nodes:
+            return
+        if not is_tree:
+            raise ValueError(
```

Comment:
Should be something derived from DvcException, so that it is caught in run() and not presented as an `unexpected error`.

Reply:
Done in commit: https://github.com/iterative/dvc/pull/1781/commits/263dee8650aca16315a3713fb8e83919fecffa40